### PR TITLE
Nets and Primitives Refactoring

### PR DESCRIPTION
**Pull request by maxcapodi78**
Now Primitives and Nets from EDB are casted into a new data object. They both inherit the original EDB object so the functionalities and back compatibility is preserved.

### CODE REVIEW

File: pyaedt/edb_core/EDB_Data.py [hunk 1]
```diff
@@ -13,6 +16,431 @@
...
+    >>> edb_net.name # Class Property
+    >>> edb_net.GetName() # EDB Object Property
+    """
+
+    def __getattr__(self, key):
+        if key in dir(self):
+            try:
+                return getattr(self, key)
+            except:
+                raise AttributeError("Attribute not present")
```
Comment (Alberto-DM) [inline pyaedt/edb_core/EDB_Data.py:37]: When is it going in this `except` considering the `if` at line 33?

**Review by Alberto-DM — APPROVED** [review]
Look good to me. 
I have only one question about the `try: except:` clause.

File: pyaedt/edb_core/EDB_Data.py [hunk 2]
```diff
@@ -13,6 +16,431 @@
...
+    #     if key in dir(self):
+    #         try:
+    #             return setattr(self, key, value)
+    #         except:
+    #             raise AttributeError("Attribute not present")
+    #     else:
+    #         try:
+    #             return setattr(self.rimitive, key, value)
+    #         except:
+    #             raise AttributeError("Attribute not present")
```
Comment (MaxJPRey) [inline pyaedt/edb_core/EDB_Data.py:55]: @maxcapodi78 Can it be deleted or otherwise remove the comments?

File: pyaedt/edb_core/layout.py [hunk 3]
```diff
@@ -92,7 +93,8 @@ def update_primitives(self):
             layoutObjectInstances = layoutInstance.GetAllLayoutObjInstances()
             for el in layoutObjectInstances.Items:
                 try:
-                    self._prims.append(el.GetLayoutObj())
+                    # self._prims.append(el.GetLayoutObj())
```
Comment (MaxJPRey) [inline pyaedt/edb_core/layout.py:96]:
```suggestion
```

File: pyaedt/edb_core/EDB_Data.py [hunk 4]
```diff
@@ -13,6 +16,431 @@
...
+
+    @aedt_exception_handler
+    def is_arc(self, point):
+        """Either if a point is an arc or not.
+
+        Returns
+        -------
+        bool
+        """
+        return not point.IsArc()
```
Comment (MaxJPRey) [inline pyaedt/edb_core/EDB_Data.py:370]: Should it be:
```suggestion
        return point.IsArc()
```
It is surprising to have the ``not``. Maybe the `IsArc()` implementation is not intuitive.

File: pyaedt/edb_core/EDB_Data.py [hunk 5]
```diff
@@ -13,6 +16,431 @@
...
+
+    @aedt_exception_handler
+    def points(self, arc_segments=6):
+        """Return the list of points with arcs converted to segments.
+
+        Parameters
+        ----------
+        arc_segments : int
+            Number of facets to convert an arc. Default is `6`.
+
```
Comment (MaxJPRey) [inline pyaedt/edb_core/EDB_Data.py:310]:
```suggestion
```

File: pyaedt/edb_core/EDB_Data.py [hunk 6]
```diff
@@ -13,6 +16,431 @@
...
+        if reverse:
+            xr.reverse()
+            yr.reverse()
+        # fmt: on
+        return xr, yr
+
+    def _get_points_for_plot(self, my_net_points, num):
+        """
+        Get the points to be plot
```
Comment (MaxJPRey) [inline pyaedt/edb_core/EDB_Data.py:277]:
```suggestion
        Get the points to be plotted.
```

File: pyaedt/edb_core/EDB_Data.py [hunk 7]
```diff
@@ -13,6 +16,431 @@
...
+            Arc height.
+        n : int
+            Number of points to generate along the arc.
+        tol : float
+            Geometric tolerance.
+
+        Returns
+        -------
+        list
+            points generated along the arc.
```
Comment (MaxJPRey) [inline pyaedt/edb_core/EDB_Data.py:221]:
```suggestion
            Points generated along the arc.
```

File: pyaedt/edb_core/EDB_Data.py [hunk 8]
```diff
@@ -13,6 +16,431 @@
...
+        h : float
+            Arc height.
+        n : int
+            Number of points to generate along the arc.
+        tol : float
+            Geometric tolerance.
+
+        Returns
+        -------
+        list
```
Comment (MaxJPRey) [inline pyaedt/edb_core/EDB_Data.py:220]: @maxcapodi78 I think that it returns 2 lists.

File: pyaedt/edb_core/EDB_Data.py [hunk 9]
```diff
@@ -13,6 +16,431 @@
...
+    def _get_points_for_plot(self, my_net_points, num):
+        """
+        Get the points to be plot
+        """
+        # fmt: off
+        x = []
+        y = []
+        for i, point in enumerate(my_net_points):
+            # point = my_net_points[i]
+            if not point.IsArc():
```
Comment (MaxJPRey) [inline pyaedt/edb_core/EDB_Data.py:284]: You can use the `is_arc` property (defined below) here.

**Review by MaxJPRey — APPROVED** [review]

File: pyaedt/edb_core/EDB_Data.py (hunk 1 above)
Comment (maxcapodi78) [reply pyaedt/edb_core/EDB_Data.py:37]: when you pass a wrong argument

File: pyaedt/edb_core/EDB_Data.py (hunk 2 above)
Comment (maxcapodi78) [reply pyaedt/edb_core/EDB_Data.py:55]: Done. Thanks

**Pull request merged**